### ISSUE TITLE
Docs

### ISSUE THREAD

**Issue by MartinOpde** [Nethereum/Nethereum.Uport#1]
Do you have docs for uPort integration ?

**Comment by juanfranblanco** [comment]
Oh Uport has changed still work in progress, so I am waiting for it to settle to integrate.

**Comment by MartinOpde** [comment]
Do you know if I can integrate uport within an app using Nethereum ?

**Comment by juanfranblanco** [comment]
Well when is ready, it will be possible.

**Comment by pvanroos** [comment]
@juanfranblanco Hey.  I would love to use Nethereum with uPort.  Any progress on this?  Is there a better option than uPort for auth?

**Comment by juanfranblanco** [comment]
Hi @pvanroos  well the idea is to start working on a DID implementation that will integrate with uports registry and provide a register. What is your use case for authentication? What is your device? A simple authentication is to integrate with the browser signer (Metamask :) ) using blazor either server side or client side which i am also working on

**Comment by pvanroos** [comment]
Hi @juanfranblanco 

I'm with ya man.  I'm building pretty much the same thing. 

...basically a Blazor WebAssembly app, served from IPFS (via IPNS), auth'd with some sort of DID.  Then I use some sort of private key from the DID to sign all of the other important stuff like cryptocurrency, data, etc.  

This is my dream.  I've been trying for years to build something like this.  Fully decentralized.  

Don't get me wrong. I'm not a decentralized bigot.  I understand the pluralism needed in different parts of an app.  Some parts are centralized, distributed and decentralized.  It's not a purity contest.  It all depends on what the requirements are.

Anyway, Nethereum rocks and I'd love to be able to get something working that is actually usable by my father.

**Comment by juanfranblanco** [comment]
@pvanroos Well we have now a full release of Blazor web assembly and the DID standard although rather abstracted, seems to be there, now is a question of implementation although there is plenty to work there as there is nothing in .Net. The simplest starting point is to create the Ethereum registry and resolver (my plan) and the move up to the Universal ones. 

What is your expectation for authentication / authorization using a dapp?

**Comment by juanfranblanco** [comment]
And I share your goals :) Anybody should be able to "simply" use it,  "simply" develop an app with it, and chose if they want to host their DID Documents themselves or delegate them to a third party. Ideally the former first and the second will be a plugin to existing or new providers.

**Comment by pvanroos** [comment]
@juanfranblanco Exactly.  You are more eloquent and articulate at explaining it.  :-)

Are there any alternatives to uPort that you've come across?

**Comment by juanfranblanco** [comment]
Ethereum specific, most build on the standards set by Uport, https://w3c-ccg.github.io/did-method-registry/ although there is a java driver for https://github.com/decentralized-identity/uni-resolver-driver-did-erc725 which is another standard of identity in ethereum.

**Comment by juanfranblanco** [comment]
oh and a sidetree for ethereum, which just pushes txn events https://github.com/decentralized-identity/sidetree-ethereum for immutability but keeps the data internally.
